### PR TITLE
Adding tswift fan club example

### PR DESCRIPTION
**Pull request by nitro-neal**
# Fan Club Credential Example
## Overview
This example demonstrates how to use the @web5/credentials package to issue and verify Verifiable Credentials (VCs). It includes creating issuer and subject DIDs, issuing a signed VC, and then verifying it. Also it will show how to issue a self signed VC and store it in your DWN.

## Prerequisites
* Node.js v18.16.1 or later
* npm (Node Package Manager)

## Installation
Clone the tutorial repository and navigate to the `fan-club-credential` directory:

```bash
git clone https://github.com/TBD54566975/developer.tbd.website.git
cd developer.tbd.website/examples/tutorials/fan-club-credential
```

Install the required npm packages:
```bash
npm install
```

## Usage
The index.js file contains the main logic for the example. It performs the following steps:

* Creates issuer and subject DIDs.
* Defines a ticket credential class (TSwiftFanClub).
* Creates an unsigned VC.
* Signs the VC using the issuer's did.
* Verifies the signed VC.

Run the script using Node.js:

```bash
node index.js
```

### CODE REVIEW

**Comment by blackgirlbytes** [comment]
Thanks Neal! I think before merging our team needs to add a step by step tutorial similar to:
- https://developer.tbd.website/docs/web5/build/apps/shared-todo-app
- https://developer.tbd.website/docs/web5/build/apps/dinger-tutorial

and we may need to add tests

File: examples/tutorials/fan-club-credential/index.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+ */
+
+// Prerequisites: Create issuer (fan club issuer)
+const fanClubIssuerDid = await DidKeyMethod.create();
+
+// Prerequisites: Create subject (alice)
+const aliceDid = await DidKeyMethod.create();
+
+// Create new credential
+class TSwiftFanClub {
```
[the remark this answers is not in the document]
Comment (nitro-neal) [reply examples/tutorials/fan-club-credential/index.js:23]: Ok!

**Review by angiejones — APPROVED** [review]

**Review by dayhaysoos — APPROVED** [review]

File: examples/tutorials/fan-club-credential/index.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
+/** This is an example of how to use the web5/credentials package:  https://www.npmjs.com/package/@web5/credentials */
+
+import { VerifiableCredential, PresentationExchange } from '@web5/credentials';
+import { DidKeyMethod, utils as didUtils } from '@web5/dids';
+import { Ed25519 } from '@web5/crypto';
+import { Web5 } from '@web5/api';
+
+// From: https://developer.tbd.website/docs/web5/quickstart
```
Comment (angiejones) [inline examples/tutorials/fan-club-credential/index.js:8]:
```suggestion

```

File: examples/tutorials/fan-club-credential/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Install the required npm packages:
+```bash
+npm install
+```
+
+## Usage
+The index.js file contains the main logic for the example. It performs the following steps:
+
+* Creates issuer and subject DIDs.
+* Defines a ticket credential class (SwiftiesFanClub).
```
Comment (angiejones) [inline examples/tutorials/fan-club-credential/README.md:26]:
```suggestion
* Defines a fan club credential class (SwiftiesFanClub).
```

File: examples/tutorials/fan-club-credential/index.js [hunk 4]
```diff
@@ -26,41 +25,28 @@ class SwiftiesFanClub {
   }
 }
 // Credential needs: type, issuer, subject, data
-const vc = VerifiableCredential.create({type: 'SwiftiesFanClub', issuer: fanClubIssuerDid.did, subject: aliceDid.did, data: new SwiftiesFanClub('#1 Fan', true)});
+const vc = await VerifiableCredential.create({ type: 'SwiftiesFanClub', issuer: fanClubIssuerDid.did, subject: aliceDid.did, data: new SwiftiesFanClub('#1 Fan', true) });
```
Comment (angiejones) [inline examples/tutorials/fan-club-credential/index.js:28]:
```suggestion
const vc = await VerifiableCredential.create({ 
                type: 'SwiftiesFanClub', 
                issuer: fanClubIssuerDid.did, 
                subject: aliceDid.did, 
                data: new SwiftiesFanClub('#1 Fan', true) 
            });
```

File: examples/tutorials/fan-club-credential/index.js [hunk 5]
```diff
@@ -26,41 +25,28 @@ class SwiftiesFanClub {
...
-// TODO: This is going away soon
-const signOptions = {
-  issuerDid  : fanClubIssuerDid.did,
-  subjectDid : aliceDid.did,
-  kid        : `${fanClubIssuerDid.did}#${fanClubIssuerDid.did.split(':')[2]}`,
-  signer     : async (data) => await Ed25519.sign({ data, key: privateKeyJwk })
-};
-
-const signedVcJwt = await vc.sign(signOptions);
+const signedVcJwt = await vc.sign({ did: fanClubIssuerDid });
```
Comment (angiejones) [inline examples/tutorials/fan-club-credential/index.js:32]: nice!

**Comment by angiejones** [comment]
> Thanks Neal! I think before merging our team needs to add a step by step tutorial similar to:
> 
> * https://developer.tbd.website/docs/web5/build/apps/shared-todo-app
> * https://developer.tbd.website/docs/web5/build/apps/dinger-tutorial
> 
> and we may need to add tests

Added a corresponding tutorial, although it's not an exact match. I changed the order of the steps (moving `verify` to last), and omitted the use of user agents and dwn records

**Review by angiejones — APPROVED** [review]

**Comment by angiejones** [comment]
blocked by #1073

**Comment by angiejones** [comment]
Not sure why tests are failing. the package lock looks fine. I cloned the repo, deleted it, and re-ran `npm install` to generate a new one. it was exactly the same as the one already committed

https://github.com/TBD54566975/developer.tbd.website/actions/runs/7282017724/job/19843612695?pr=1056#step:5:12

**Comment by angiejones** [comment]
> blocked by #1073

unblocked now

**Comment by acekyd** [comment]
This now works on CSB, no tests added - similar to the book reviews but I think the branch is outdated hence the failed tests

**Comment by angiejones** [comment]
> This now works on CSB, no tests added - similar to the book reviews but I think the branch is outdated hence the failed tests

Ace, I updated the branch. There's an error with the lock file. Can you look into fixing this?

image

**Comment by acekyd** [comment]
Looking all good. Not sure but do we need to manually start the netlify preview? Seems to be the only thing left. @angiejones

**Review by angiejones — APPROVED** [review]

**Pull request merged**